src/org/traccar/database/PermissionsManager.java

```diff
@@ -395,4 +397,32 @@ public User getUserByToken(String token) {
         return users.get(usersTokens.get(token));
     }
 
+    public Object lookupPreference(long userId, String key, Object defaultValue) {
+        String methodName = "get" + key.substring(0, 1).toUpperCase() + key.substring(1);
+        Object preference;
+        Object serverPreference = null;
+        Object userPreference = null;
+        try {
+            Method method = null;
+            method = User.class.getMethod(methodName, (Class<?>[]) null);
+            if (method != null) {
+                userPreference = method.invoke(users.get(userId), (Object[]) null);
+            }
+            method = null;
+            method = Server.class.getMethod(methodName, (Class<?>[]) null);
+            if (method != null) {
+                serverPreference = method.invoke(server, (Object[]) null);
+            }
+        } catch (NoSuchMethodException | SecurityException | IllegalAccessException
```

Comment:
Most of the exceptions can be combined into `ReflectiveOperationException`. Everything else looks fine.